containers/daap-python-base/src/var/task/versioning.py

```diff
@@ -141,6 +175,61 @@ def metadata_update_type(data_product_metadata) -> UpdateType:
         return UpdateType.MinorUpdate
 
 
+def schema_update_type(
+    data_product_schema: DataProductSchema,
+) -> tuple[UpdateType, dict]:
+    """
+    Figure out whether changes to the input data represent a major or minor schema update
+    and return the changes as a dict, e.g.
+        {table_name: {columns:{...}, non_column_fields: {...}}}
+    """
+    if not data_product_schema.exists or not data_product_schema.valid:
+        return UpdateType.NotAllowed, {}
+
+    changed_fields = data_product_schema.changed_fields()
```

Comment:
Added this in because it's likely data captured at that level for schema could increase and then we can just add into `MINOR_UPDATE_SCHEMA_FIELDS` or even a `MAJOR_UPDATE_SCHEMA_FIELDS`